tests/e2e/Services/Databases/DatabasesBase.php

```diff
@@ -4216,4 +4216,82 @@ public function testUpdateWithExistingRelationships(array $data): void
 
         $this->assertEquals(200, $update['headers']['status-code']);
     }
+
+    /**
+     * @depends testCreateDatabase
+     */
+    public function testDeleteRelationAttributeIfRelatedCollectionDeleted(array $data): void
+    {
+        $databaseId = $data['databaseId'];
+
+        // Create level 1 collection
+        $level1 = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'collectionId' => ID::unique(),
+            'name' => 'Level 1',
+            'documentSecurity' => true,
+            'permissions' => [
+                Permission::create(Role::user($this->getUser()['$id'])),
+                Permission::read(Role::user($this->getUser()['$id'])),
+            ],
+        ]);
+        // Create level 2 collection
+        $level2 = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'collectionId' => ID::unique(),
+            'name' => 'Level 2',
+            'documentSecurity' => true,
+            'permissions' => [
+                Permission::create(Role::user($this->getUser()['$id'])),
+                Permission::read(Role::user($this->getUser()['$id'])),
+            ],
+        ]);
+
+        // Create relationship
+        $response = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections/' . $level1['body']['$id'] . '/attributes/relationship', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'relatedCollectionId' => $level2['body']['$id'],
+            'type' => Database::RELATION_MANY_TO_ONE,
+            'twoWay' => false,
+            'key' => 'level2',
+            'onDelete' => Database::RELATION_MUTATE_CASCADE,
+        ]);
+
+        $this->assertEquals(202, $response['headers']['status-code']);
+        $this->assertEquals(false, $response['body']['array']);
+        $this->assertEquals('level2', $response['body']['key']);
+        $this->assertEquals('cascade', $response['body']['onDelete']);
+        $this->assertEquals($level2['body']['$id'], $response['body']['relatedCollection']);
+        $this->assertEquals('manyToOne', $response['body']['relationType']);
+        $this->assertEquals(false, $response['body']['required']);
+        $this->assertEquals(false, $response['body']['twoWay']);
+        $this->assertEquals('relationship', $response['body']['type']);
+
+        // Delete level 2 collection
```

Comment:
let's only keep line break and no comment, seems straight forward

Reply:
Done